### DISCUSSION TITLE
v2.4.0

### DISCUSSION THREAD

**Discussion by buchdag** [nginx-proxy/acme-companion#1125] [Announcements]
## What's Changed

⚠️ This release is not compatible out of the box with [versions of nginx-proxy](https://github.com/nginx-proxy/nginx-proxy/releases) below `1.6` (see #1123 and https://github.com/nginx-proxy/nginx-proxy/pull/2446 / https://github.com/nginx-proxy/nginx-proxy/pull/2468 / https://github.com/nginx-proxy/nginx-proxy/pull/2472).

If you use acme-companion >= `2.4`, either upgrade nginx-proxy to >= `1.6` or use the `ACME_HTTP_CHALLENGE_LOCATION` environment variable introduced in #1123 to re-enable challenge location handling by acme-companion.

### Features

- feat: disable automatic ACME HTTP challenge location configuration by @buchdag in https://github.com/nginx-proxy/acme-companion/pull/1123

### Dependencies

- build: bump alpine from 3.19.1 to 3.20.0 by @dependabot in https://github.com/nginx-proxy/acme-companion/pull/1119
- build: bump nginxproxy/docker-gen from 0.12.1 to 0.14.0 by @dependabot in https://github.com/nginx-proxy/acme-companion/pull/1111, https://github.com/nginx-proxy/acme-companion/pull/1118 and https://github.com/nginx-proxy/acme-companion/pull/1121

### Other changes

- test: upgrade pebble to 2.5.2 by @buchdag in https://github.com/nginx-proxy/acme-companion/pull/1120
- test: upgrade pebble to 2.6.0 by @buchdag in https://github.com/nginx-proxy/acme-companion/pull/1124

**Full Changelog**: https://github.com/nginx-proxy/acme-companion/compare/v2.3.0...v2.4.0

<hr /><em>This discussion was created from the release <a href='https://github.com/nginx-proxy/acme-companion/releases/tag/v2.4.0'>v2.4.0</a>.</em>

**Comment by jolly-jump** [comment]
Hi.
Just to let you know: I have let watchtower upgrade my setup 
```
nginxproxy/nginx-proxy:latest
nginxproxy/acme-companion:latest
```
The setup is old, so no guarantee the error is on my side using some old configuration.

However, using the companion image: 3e2122a42d84 from 8 days ago, I ran into trouble / rate limit since no certificate would update.
Waiting the rate limit period and using the image:  a044bebb8a7b from 2 weeks ago, everything went back to normal.

I could glimpse at the logs and it had problems with curl not having some option available or something like this.

EDIT: I did try the current container once more. All the certificates that are up to date (of course) do not cause trouble. Adding a container which would update its cert, it says:
```
proxy-le  | 2024/06/15 19:36:15 Received event start for container 312599b49273
proxy-le  | 2024/06/15 19:36:20 Debounce minTimer fired
proxy-le  | 2024/06/15 19:36:20 Generated '/app/letsencrypt_service_data' from 18 containers
proxy-le  | 2024/06/15 19:36:20 Running '/app/signal_le_service'
proxy-le  | Creating/renewal api.test.mydomain.org certificates... (api.test.mydomain.org)
proxy-le  | [Sat Jun 15 19:36:21 UTC 2024] Using CA: https://acme-v02.api.letsencrypt.org/directory
proxy-le  | [Sat Jun 15 19:36:21 UTC 2024] Creating domain key
proxy-le  | [Sat Jun 15 19:36:21 UTC 2024] The domain key is here: /etc/acme.sh/admin@mydomain.org/api.test.mydomain.org/api.test.mydomain.org.key
proxy-le  | [Sat Jun 15 19:36:21 UTC 2024] Generate next pre-generate key.
proxy-le  | [Sat Jun 15 19:36:21 UTC 2024] Single domain='api.test.mydomain.org'
proxy-le  | [Sat Jun 15 19:36:21 UTC 2024] Getting domain auth token for each domain
proxy-le  | [Sat Jun 15 19:36:23 UTC 2024] Getting webroot for domain='api.test.mydomain.org'
proxy-le  | [Sat Jun 15 19:36:23 UTC 2024] Verifying: api.test.mydomain.org
proxy-le  | [Sat Jun 15 19:36:24 UTC 2024] Pending, The CA is processing your order, please just wait. (1/30)
proxy-le  | [Sat Jun 15 19:36:27 UTC 2024] Invalid status, api.test.mydomain.org:Verify error detail:X.X.X.X: Invalid response from http://api.test.mydomain.org/.well-known/acme-challenge/XYZ...Arsg: 502
proxy-le  | [Sat Jun 15 19:36:27 UTC 2024] Please check log file for more details: /dev/null
```
replaced my domain with mydomain and cropped the acme-challenge...

EDIT: as soon as I go back to said image 2 weeks ago, the verification runs just fine.

Sorry for being not more specific, lets hope this is only on my system.
Since the system ran absolutely fine for several years, a big thank you for your service!

2nd EDIT: found a newer image, where it was/is still working
